src/lib/atemSocket.ts

```diff
@@ -34,29 +21,17 @@ export class AtemSocket extends EventEmitter {
 		this._debug = options.debug || false
 		this.log = options.log || this.log
 
-		this._createSocket()
+		this._createSocketProcess()
+
+		process.on('exit', () => {
+			if (this._socketProcess) {
+				this._socketProcess.removeAllListeners()
```

Comment:
We need to recreate the Socket process, possibly emit errors as well. And I guess resend commands that have not been acknowledged yet.

Reply:
This particular line is about when the _parent_ process exits. When our parent begins exiting, we need to remove our error handlers from the child. If we don't, our error handlers will throw an error every time we exit the program.

You're looking for https://github.com/nrkno/tv-automation-atem-connection/pull/19/files/3ce0992e68a4c03ef4cf1bcac2c52cba86d95417#diff-e7a164ad6407add743a708771e7eecb8R97

Reply:
Whoopsie, you're right. All these sockets and connections everywhere ;)

Reply:
Yeah a resend mechanism would probably be good. I'll evaluate how complex it will be to implement.